File: Problems/Codeforce/Contest/Codeforces Round 1002 (Div. 2)/c.py
def solve():
    n = int(input())
    a = [list(map(int, input().split())) for _ in range(n)]

    pr = [[0] * (n + 1) for _ in range(n)]

    for i in range(n):
        for j in range(n):
            pr[i][j + 1] = pr[i][j] + a[i][n - 1 - j]

    ans = -1
    mx = n

    for id in range(mx):
        cnt = 0
        for i in range(n):
            if pr[i][id] == id:
                cnt += 1
        
        if cnt != 0:
            ans = id
        
        mx = min(mx, id + cnt)

    print(ans + 1)
t=int(input())
for _ in range(t):
    solve()

File: Problems/Codeforce/Contest/Codeforces Round 1002 (Div. 2)/c.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

void solve() {
    int n;
    cin >> n;

    vector<vector<int>> a(n, vector<int>(n));
    vector<vector<int>> pr(n, vector<int>(n + 1, 0));

    // Read input matrix
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cin >> a[i][j];
        }
    }

    // Compute prefix sum array
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            pr[i][j + 1] = pr[i][j] + a[i][n - 1 - j];
        }
    }

    int ans = -1;
    int mx = n;

    for (int id = 0; id < mx; id++) {
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            if (pr[i][id] == id) {
                cnt++;
            }
        }

        if (cnt != 0) {
            ans = id;
        }

        mx = min(mx, id + cnt);
    }

    cout << ans + 1 << "\n";
}

int main() {
    int t=0;
    cin>>t;
    for(int i=0;i<t;i++){
    solve();
    }

    return 0;
}
